libbeat/reader/readjson/docker_json.go

```diff
@@ -107,11 +106,12 @@ func (p *DockerJSONReader) parseCRILog(message *reader.Message, msg *logLine) er
 		"stream": msg.Stream,
 	})
 	// Remove ending \n for partial messages
-	message.Content = []byte(log[i])
+	message.Content = log[i]
 	if partial {
-		message.Content = bytes.TrimRightFunc(message.Content, func(r rune) bool {
-			return r == '\n' || r == '\r'
-		})
+		l := len(message.Content)
+		if l > 0 && message.Content[l] == '\n' || message.Content[l] == '\r' {
+			message.Content = message.Content[:l-1]
+		}
```

Comment:
this condition is not doing the same thing as TrimRightFunc, as it would not trim `\r\n`

Reply:
I'm on a 100% linux systems with no \r. Is that for windows?